src/unix/mod.rs

```diff
@@ -320,6 +320,24 @@ pub const ATF_PERM: ::c_int = 0x04;
 pub const ATF_PUBL: ::c_int = 0x08;
 pub const ATF_USETRAILERS: ::c_int = 0x10;
 
+pub const FNM_PERIOD: c_int = 1 << 2;
+pub const FNM_CASEFOLD: c_int = 1 << 4;
+pub const FNM_NOMATCH: c_int = 1;
+
+cfg_if! {
+    if #[cfg(any(
+        target_os = "macos",
+        target_os = "freebsd",
+        target_os = "android",
+    ))] {
+        pub const FNM_PATHNAME: c_int = 1 << 1;
+        pub const FNM_NOESCAPE: c_int = 1 << 0;
+    } else {
+        pub const FNM_PATHNAME: c_int = 1 << 0;
+        pub const FNM_NOESCAPE: c_int = 1 << 1;
+    }
+}
```

Comment:
Overall this looks pretty straightforward, but do you have links to headers defining these? I am a bit surprised they are opposite on a few OSs, we should make sure that's documented in the PR.

Reply:
Certainly,

<table>
<tr>
<th>

`macos`

</th>
<td>

<https://opensource.apple.com/source/Libc/Libc-825.40.1/include/fnmatch.h.auto.html>

</td>
<td>

```c
#define	FNM_NOESCAPE	0x01	/* Disable backslash escaping. */
#define	FNM_PATHNAME	0x02	/* Slash must be matched by slash. */
```

</td>
</tr>

<tr>
<th>

`freebsd`

</th>
<td>

<https://github.com/freebsd/freebsd-src/blob/4a983f05d98ba1c5eebe65d29dd061dc3eab7594/include/fnmatch.h#L4>

</td>
<td>

```c
#define	FNM_NOESCAPE	0x01	/* Disable backslash escaping. */
#define	FNM_PATHNAME	0x02	/* Slash must be matched by slash. */
```

</td>
</tr>

<tr>
<th>

`android`

</th>
<td>

<https://cs.android.com/android/platform/superproject/main/+/main:bionic/libc/include/fnmatch.h;drc=bbd39aa8e66dfe325732934d1aeb0da52b4750d5>

</td>
<td>

```c
/** fnmatch() flag to disable backslash escaping. */
#define FNM_NOESCAPE     0x01 
/** fnmatch() flag to ensure that slashes must be matched by slashes. */
#define FNM_PATHNAME     0x02
```

</td>
</tr>
<tr>
<th>

`musl`

</th>
<td>

<https://git.musl-libc.org/cgit/musl/tree/include/fnmatch.h?id=03860303c298f2ec7a9c0de683a68f4ab61697f0>

</td>
<td>

```c
#define	FNM_PATHNAME 0x1
#define	FNM_NOESCAPE 0x2
```

</td>
</tr>
<tr>
<th>

`glibc`

</th>
<td>

<https://elixir.bootlin.com/glibc/glibc-2.40.9000/source/posix/fnmatch.h#L33>


</td>
<td>

```c
#define	FNM_PATHNAME	(1 << 0) /* No wildcard can ever match `/'.  */
#define	FNM_NOESCAPE	(1 << 1) /* Backslashes don't quote special chars.  */
```

</td>
</tr>
</table>

Reply:
Awesome, thanks for the sources!